code/game/machinery/kitchen/microwave.dm

```diff
@@ -182,17 +183,31 @@
 	user.set_machine(src)
 	if(broken > 0)
 		to_chat(user, "<span class='warning'>\The [name] is broken! You'll need to fix it before using it.</span>")
-	else if(dirty == 100)
+	else if(dirty >= 100)
 		to_chat(user, "<span class='warning'>\The [name] is dirty! You'll need to clean it before using it.</span>")
 	else
 		ui_interact(user)
 
+/obj/machinery/microwave/examine(var/mob/user)
+	..()
+	if(broken > 0)
+		to_chat(user, "It's broken!")
+	else if(dirty >= 100)
+		to_chat(user, "The insides are completely filthy!")
+	else if(dirty > 75)
+		to_chat(user, "It's covered in stains.")
+	else if(dirty > 50)
+		to_chat(user, "It's pretty messy.")
+	else if(dirty > 25)
+		to_chat(user, "It's a bit dirty.")
+
 /*******************
 *   Microwave Menu
 ********************/
 
 VUEUI_MONITOR_VARS(/obj/machinery/microwave, microwavemonitor)
 	watch_var("operating", "on", CALLBACK(null, .proc/transform_to_boolean, FALSE))
+	watch_var("recipe", "recipe")

```

Comment:
You need to use too boolean transform here.

Reply:
Yeah, you're right. I had removed it as a test and it seemed to fix the issue, but that was just a placebo it seems like and my issue is actually `recipe` not being sent to state before the progress bar area is drawn.